### PR TITLE
`azurerm_data_factory_dataset_delimited_text`: Expose `compression_codec`

### PR DESCRIPTION
**Pull request by brunhil**
Expose Compression Codec as requested in #9980

### CODE REVIEW

File: azurerm/internal/services/datafactory/data_factory_dataset_delimited_text_resource_test.go [hunk 1]
```diff
@@ -59,7 +60,8 @@ func TestAccDataFactoryDatasetDelimitedText_http_update(t *testing.T) {
 				check.That(data.ResourceName).Key("schema_column.#").HasValue("2"),
 				check.That(data.ResourceName).Key("additional_properties.%").HasValue("1"),
 				check.That(data.ResourceName).Key("description").HasValue("test description 2"),
-				check.That(data.ResourceName).Key("compression_level").HasValue("Fastest"),
+				check.That(data.ResourceName).Key("compression_codec").HasValue("lz4"),
+				check.That(data.ResourceName).Key("compression_level").HasValue("Optimal"),
```
Comment (manicminer) [inline azurerm/internal/services/datafactory/data_factory_dataset_delimited_text_resource_test.go:64]: `compression_level` is "Fastest" in the config here
```suggestion
				check.That(data.ResourceName).Key("compression_level").HasValue("Fastest"),
```

**Review by manicminer — CHANGES_REQUESTED** [review]
Thanks for this @brunhil. There's just one change needed in a test check, otherwise this LGTM.

**Review by mbfrahry — APPROVED** [review]
LGTM!

**Comment by katbyte** [comment]
linting failure seems spurious, merging

**Pull request merged**